clang/test/CodeGenSYCL/stall_enable.cpp

```diff
@@ -0,0 +1,32 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm -o - %s | FileCheck %s
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+queue q;
+
+[[intel::stall_enable]] void test() {}
+
+class Foo {
+public:
+  [[intel::stall_enable]] void operator()() const {}
+};
+
+int main() {
+  q.submit([&](handler &h) {
+    Foo f;
+    h.single_task<class test_kernel1>(f);
+
+    h.single_task<class test_kernel2>(
+        []() { test(); });
+
+    h.single_task<class test_kernel3>(
+        []() [[intel::stall_enable]]{});
+  });
+  return 0;
+}
+
+// CHECK: define spir_kernel void @"{{.*}}test_kernel1"() #0 {{.*}} !stall_enable [[FOO:![0-9]+]]
+// CHECK: define spir_kernel void @"{{.*}}test_kernel2"() #0 {{.*}} !stall_enable [[FOO:![0-9]+]]
+// CHECK: define spir_kernel void @"{{.*}}test_kernel3"() #0 {{.*}} !stall_enable [[FOO:![0-9]+]]
```

Comment:
Do we expect the same metadata for all 3 kernels? If yes, we can do
```suggestion
// CHECK: define spir_kernel void @"{{.*}}test_kernel2"() #0 {{.*}} !stall_enable [[FOO]]
// CHECK: define spir_kernel void @"{{.*}}test_kernel3"() #0 {{.*}} !stall_enable [[FOO]]
```

Reply:
@AlexeySotkin, Thanks for taking a look at the patch and the feedback. All three kernels expect same metadata here. I have updated test.